compiler/rustc_target/src/spec/aarch64_unknown_linux_dynmusl.rs

```diff
@@ -0,0 +1,11 @@
+use crate::spec::Target;
+
+pub fn target() -> Target {
+    let mut base = super::aarch64_unknown_linux_musl::target();
+
+    base.llvm_target = "aarch64-unknown-linux-musl".to_string();
+    base.options.crt_static_default = false;
+    base.options.static_position_independent_executables = true;
```

Comment:
- `base.options.static_position_independent_executables` is default to `true` for linux_base
- `base.pre_link_objects_fallback`, `base.post_link_objects_fallback` and `base.crt_objects_fallback` should not set on dynmusl target ( host toolchain provide those object files ).

Reply:
Its fine to set the `pie` explicitly here, but the `link_objects` should definitely be removed as that's only necessary for self-contained targets (`musl`)

Reply:
As for the shipping of the objects, I think you'll just need to add a dash [here](https://github.com/rust-lang/rust/blob/cecdb181ade91c0a5ffab9a148dba68fc7d00ee3/src/bootstrap/compile.rs#L180)

Reply:
Should be resolved by c41347cbc62.